micall/tests/test_contig_stitcher.py

```diff
@@ -0,0 +1,1744 @@
+import random
+import logging
+import os
+import pytest
+
+import micall.core.contig_stitcher as stitcher
+from micall.core.contig_stitcher import (
+    split_contigs_with_gaps,
+    stitch_contigs,
+    GenotypedContig,
+    merge_intervals,
+    find_covered_contig,
+    stitch_consensus,
+    calculate_concordance,
+    align_all_to_reference,
+    AlignedContig,
+    disambiguate_concordance,
+    lstrip,
+    rstrip,
+)
+from micall.core.plot_contigs import plot_stitcher_coverage
+from micall.tests.utils import MockAligner, fixed_random_seed
+from micall.tests.test_denovo import check_hcv_db  # activates the fixture
+from micall.tests.test_remap import load_projects  # activates the "projects" fixture
+
+
+logging.getLogger("micall.core.contig_stitcher").setLevel(logging.DEBUG)
+logging.getLogger("micall.core.plot_contigs").setLevel(logging.DEBUG)
+
+
+random.choice([check_hcv_db, load_projects])  # make linters not complain about unused imports.
+
+
+@pytest.fixture()
+def exact_aligner(monkeypatch):
+    monkeypatch.setattr("micall.core.contig_stitcher.Aligner", MockAligner)
+
+
+@pytest.fixture
+def visualizer(request, tmp_path):
+    stitcher.context.set(stitcher.StitcherContext())
+    test_name = request.node.name
+    plot_name = test_name + ".svg"
+    pwd = os.path.dirname(__file__)
+    plots_dir = os.path.join(pwd, "data", "stitcher_plots")
+    os.makedirs(plots_dir, exist_ok=True)
+    path_to_expected = os.path.join(plots_dir, plot_name)
+    path_to_produced = os.path.join(tmp_path, plot_name)
+
+    def check():
+        logs = stitcher.context.get().events
+        figure = plot_stitcher_coverage(logs, path_to_produced)
+
+        with open(path_to_produced, "r") as produced_file:
+            produced_data = produced_file.read()
+        with open(path_to_expected, "r") as expected_file:
+            expected_data = expected_file.read()
+
+        assert produced_data == expected_data, (
+            "The contents of the stitched contigs plot"
+            " does not match the expected contents."
+        )
+
+        return figure
+
+    return check
+
+
+def test_identical_stitching_of_one_contig(exact_aligner, visualizer):
+    # Scenario: When stitching one contig, it remains the same.
+
+    contigs = [
+        GenotypedContig(
+            name="a",
+            seq="ACTGACTG" * 100,
+            ref_name="testref",
+            group_ref="testref",
+            ref_seq="T" * 20 + "ACTGACTG" * 110 + "T" * 20,
+            match_fraction=1.0,
+        ),
+    ]
+
+    results = list(stitch_contigs(contigs))
+    assert len(results) == 1
+    assert results[0].seq == contigs[0].seq
+
+    assert len(visualizer().elements) > len(contigs)
```

Comment:
Yes, that is true. This was a convenience thing for me when I was regenerating the test cases often - this check would fail if the regenerated example does not make sense.

Should I remove this check completely?

Reply:
My point is that it couldn't fail. If it didn't have the expected number of SVG elements, then `visualizer()` would raise an `AssertionError`, right? Are you worried that you would generate a test case with empty SVG? If so, could you put an `assert` in the code that generates the test cases and remove this one?